packages/appstore/src/components/wallet-header/wallet-header.tsx

```diff
@@ -5,69 +5,64 @@ import WalletCurrencyCard from './wallet-currency-card';
+import { observer, useStore } from '@deriv/stores';
 import './wallet-header.scss';
 
 type TWalletHeader = {
-    account_type: TAccountCategory;
-    shortcode?: TWalletShortcode;
-    currency?: TWalletCurrency;
-    balance?: string;
-    account_status?: TAccountStatus;
-    is_open_wallet: boolean;
-    setIsOpen: (is_open: boolean) => void;
+    wallet_account: TWalletAccount;
 };
 
-const WalletHeader = React.memo(
-    ({
-        account_status = '',
-        balance = '0.00',
-        currency = 'USD',
-        shortcode = 'svg',
-        account_type = 'real',
-        is_open_wallet,
-        setIsOpen,
-    }: TWalletHeader) => {
-        const is_demo = account_type === 'demo';
+const WalletHeader = observer(({ wallet_account }: TWalletHeader) => {
+    const { client, traders_hub } = useStore();
+    const { switchAccount, loginid } = client;
+    const is_active = loginid === wallet_account.loginid;
+    const { multipliers_account_status } = traders_hub;
 
-        const wallet_btns = getWalletHeaderButtons(is_demo);
+    const is_demo = wallet_account.is_virtual;
 
-        const onArrowClickHandler = () => {
-            setIsOpen(!is_open_wallet);
-        };
+    const wallet_buttons = getWalletHeaderButtons(is_demo);
 
-        return (
-            <div
-                className={classNames('wallet-header', {
-                    'wallet-header__demo': is_demo,
-                })}
-            >
-                <div className='wallet-header__container'>
-                    <WalletCurrencyCard account_type={account_type} currency={currency} />
-                    <div className='wallet-header__description'>
-                        <WalletHeaderTitle is_demo={is_demo} currency={currency} shortcode={shortcode} />
-                        <WalletHeaderButtons
-                            is_disabled={!!account_status}
-                            is_open={is_open_wallet}
-                            btns={wallet_btns}
-                        />
-                    </div>
-                    <div className='wallet-header__balance'>
-                        <WalletHeaderBalance account_status={account_status} balance={balance} currency={currency} />
-                        <Icon
-                            data_testid='dt_arrow'
-                            onClick={onArrowClickHandler}
-                            icon='IcChevronDownBold'
-                            className={classNames('wallet-header__balance-arrow-icon', {
-                                'wallet-header__balance-arrow-icon-active': is_open_wallet,
-                            })}
-                        />
-                    </div>
+    const onArrowClickHandler = async () => {
+        if (loginid !== wallet_account.loginid) await switchAccount(wallet_account.loginid);
+    };
+
+    return (
+        <div className={classNames('wallet-header', { 'wallet-header__demo': wallet_account.is_demo })}>
```

Comment:
@sergei-deriv does wallet have `is_demo` key ? 
why we use `wallet_account.is_demo` and `wallet_account.is_virtual` for `is_demo` interchangeably?